helm-charts/splunk-otel-collector/templates/configmap-fluentd.yaml

```diff
@@ -289,107 +289,32 @@ data:
       {{- end }}
       {{- end }}
 
+      <filter **>
+        @type record_transformer
+        enable_ruby
+        <record>
+          com.splunk.sourcetype ${record.dig("sourcetype") ? record.dig("sourcetype") : ""}
+          com.splunk.source ${record.dig("source") ? record.dig("source") : ""}
+          com.splunk.index ${record.dig("index") ? record.dig("index") : "{{ .Values.logsBackend.hec.indexName | default "main"}}"}
+        </record>
+        remove_keys denylist,docker,kubernetes,source,sourcetype,index
+      </filter>
```

Comment:
Can we remove this extra filter and use `com.splunk.index` instead of `index` from the beginning? a follow up PR is fine

Reply:
Yep index can, source and sourcetype can't because of the `jq .record.source = ...` craziness.